File: app/src/main/java/com/example/screenorientation/MainActivity2.java
package com.example.screenorientation;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity2 extends AppCompatActivity {
    Button alertDialog;
    EditText name;
    private View view1;
    private String userName;
    private Boolean alert = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        alertDialog = (Button) findViewById(R.id.alert);

        View inflater = getLayoutInflater().inflate(R.layout.bottom_btn, null);
        name = inflater.findViewById(R.id.show);

        if (savedInstanceState != null && savedInstanceState.getBoolean("dial", false)) {
            userName = savedInstanceState.getString("name");
            name.setText(userName);
            createAlert(view1);
        }

        alertDialog.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (alert = true) {
                    createAlert(view);
                }

            }
        });
    }
    //onStop()
    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString("name", userName);
        outState.putBoolean("dial", alert);

    }
    //onCall()
    @Override
    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        savedInstanceState.getString("name",userName);
        savedInstanceState.getBoolean("dial",alert);

    }

    public void createAlert(View v) {
        AlertDialog.Builder d = new AlertDialog.Builder(MainActivity2.this);
        d.setTitle("Alert");
        final View customLayout = getLayoutInflater().inflate(R.layout.bottom_btn, null);
        d.setView(customLayout);
        EditText ed = customLayout.findViewById(R.id.show);
        toast(ed.getText().toString());
        d.show();
    }

    private void toast(String msg) {
        Toast.makeText(getApplicationContext(), msg, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/screenorientation/Activity/MainActivity2.kt
package com.example.screenorientation.Activity

import androidx.appcompat.app.AppCompatActivity
import android.widget.EditText
import android.os.Bundle
import com.example.screenorientation.R
import android.view.LayoutInflater
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AlertDialog

class MainActivity2 : AppCompatActivity(), View.OnClickListener {
    private val TAG = MainActivity2::class.java.canonicalName
    private var alertDialogButton: Button? = null
    var name: String? = null
    private var view1: View? = null
    private lateinit var ed:EditText
    var dialog: AlertDialog? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)

        alertDialogButton = findViewById<View>(R.id.alert) as Button
        dialog = AlertDialog.Builder(this).create()
        view1 = LayoutInflater.from(this).inflate(R.layout.bottom_btn, null)

        ed = view1!!.findViewById(R.id.show)
        name = ed.text.toString()
        alertDialogButton!!.setOnClickListener(this)


        if (savedInstanceState != null) {
            name = savedInstanceState.getString("name")
            ed.setText(name)
            var auth = dialog!!.onSaveInstanceState().getBoolean("alert")
            if (alert.also { auth = it }) {
                alertDialog()
            }
        }
    }

    //button on click..
    override fun onClick(view: View) {
        when (view.id) {
            R.id.alert -> if (true.also { alert = it }) {
                alertDialog()
            }

        }
    }

    //onStop()
    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        val userName = ed.text.toString()
        outState.putString("name", userName)
        outState.putBoolean("alert", alert)
    }

    //show alert
    fun alertDialog() {
        dialog!!.setTitle("Alert")
        dialog!!.setMessage("Rotate your screen")
        dialog!!.setView(view1)
        dialog!!.setCancelable(true)
        dialog!!.setOnCancelListener {
            Log.i(TAG, "onCancel: alert -> " + alert)
            alert = false
            Log.i(TAG, "onCancel: alert -> " + alert)
        }
        dialog!!.show()
    }

    //toast message if needed..
    private fun toast(msg: String) {
        Toast.makeText(applicationContext, msg, Toast.LENGTH_SHORT).show()
    }

    companion object {
        private var alert = false
    }
}